File: Java/src/Reverse_Integer.java
/**
 * Created by Kray on 2017/1/11.
 */
public class Reverse_Integer {

    public static void main(String[] args) {
        Reverse_Integer reverse_integer = new Reverse_Integer();
        System.out.println(reverse_integer.reverse(1222222222));
    }

    public int reverse(int x) {
        boolean isNeg = x < 0;
        x = x > 0 ? x : -x;
        int result = 0;
        int newResult = 0;
        while (x > 0) {
            newResult = result * 10 + x % 10;
            if ((newResult - x % 10) / 10 != result) {
                return 0;
            }
            result = newResult;
            x /= 10;
        }
        return result * (isNeg ? -1 : 1);
    }
}

File: Python/Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        is_neg = x < 0
        x = abs(x)
        while x > 0:
            res *= 10
            res += x % 10
            x -= x % 10
            x /= 10
        if is_neg:
            if res > 1 << 31:
                return 0
            else:
                return int(-1 * res)
        else:
            if res > 1 << 32 - 1:
                return 0
            else:
                return int(res)
